source/write/bulk-write.txt

```diff
@@ -0,0 +1,370 @@
+the model instance to an instance of the ``mongocxx::bulk_write`` class.
+
+The following example creates an instance of ``mongocxx::model::replace_one`` and appends
+it to a ``mongocxx::bulk_write`` instance called ``bulk``:
+
+.. literalinclude:: /includes/write/bulk-write.cpp
+   :start-after: start-bulk-replace-one
+   :end-before: end-bulk-replace-one
+   :language: cpp
+   :dedent:
+
+To replace multiple documents, you must create an instance of ``mongocxx::model::replace_one``
+for each document.
+
+.. _cpp-bulk-delete-model:
+
+Delete Operations
+~~~~~~~~~~~~~~~~~
+
+To delete a document, create an instance of the ``mongocxx::model::delete_one`` class and
+pass in a query filter specifying the document you want to delete. This model instructs
+the driver to delete only *the first* document that matches your query filter. Then, append
+the model instance to an instance of the ``mongocxx::bulk_write`` class.
+
+The following example creates an instance of ``mongocxx::model::delete_one`` and appends
+it to a ``mongocxx::bulk_write`` instance called ``bulk``:
+
+.. literalinclude:: /includes/write/bulk-write.cpp
+   :start-after: start-bulk-delete-one
+   :end-before: end-bulk-delete-one
+   :language: cpp
+   :dedent:
+
+To delete multiple documents, create an instance of the ``mongocxx::model::delete_many``
+class and pass in a query filter specifying the document you want to delete. This model
+instructs the driver to delete *all* documents that match your query filter.
+
+The following example creates an instance of ``mongocxx::model::delete_many`` and appends
+it to ``bulk``:
+
+.. literalinclude:: /includes/write/bulk-write.cpp
+   :start-after: start-bulk-delete-many
+   :end-before: end-bulk-delete-many
+   :language: cpp
+   :dedent:
+
+Run the Bulk Operation
+----------------------
+
+To run a bulk operation, call the ``execute()`` method on an instance of the 
+``mongocxx::bulk_write`` class that contains your write models. By default, the
+``execute()`` method runs the operations in the order they're appended to the
+``mongocxx::bulk_write`` instance.
+
+The following example performs the :ref:`insert <cpp-bulk-insert-model>`,
+:ref:`update <cpp-bulk-update-model>`, :ref:`replace <cpp-bulk-replace-model>`,
+and :ref:`delete <cpp-bulk-delete-model>` operations specified in the preceding
+sections of this guide by appending each corresponding write model to an instance
+of ``mongocxx::bulk_write`` and calling the ``execute()`` method. Then, it prints
+the number of modified documents:
+
+.. io-code-block::
+
+   .. input:: /includes/write/bulk-write.cpp
+      :start-after: start-bulk-run
+      :end-before: end-bulk-run
+      :language: cpp
+      :dedent:
+      :emphasize-lines: 30
+
+   .. output::
+      
+      Modified documents: 2
+
+If any of the write operations fail, the {+driver-short+} raises a
+``mongocxx::bulk_write_exception`` and does not perform any further operations.
```

Comment:
Q: Does the driver still successfully perform the operations that were ordered before the failing write operation? Or do all operations fail if just one fails/if an exception is thrown?

Reply:
Yes, the operations before the failing one still apply